* Implemented 4 ColliderData algorithms

cweiss95 opened the pull request:
* Currently have a problem with IsPointInCircle I believe

# Description
Mainly to do a practice pull request. Doesn't need to get merged just wanted to go through the motions.

So I tried to implement the 4 member functions of the ColliderData.cpp files
- 1st function check whether an AABBAData object is intersecting the current AABBData object
- 2nd function checks whether a Vector2 point is inside the current AABBData object
- 3rd function checks whether a CircleData object is inside the current CircleData object
- 4th function checks whether a Vector2 point is inside the current CircleData object

Close # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature
- [x]  Feature does not work as expected


# Checklist:

- [x] Any code I have modified/added conforms to the [Coding Standard](https://github.com/karnkaul/MicroEngine/wiki/Coding-Standards)
- [x] I have run `Tools/clang-format.sh` (Ran it after my commit but it did not change any files/require a new commit)
- [ ] My changes generate no new warnings (They have 4 warnings which I also have in Origin/Dev)
- [x] The project builds and runs without any issues
- [ ] Any dependent changes have been merged and published in downstream modules
(What does this mean?)

cweiss95 committed:
* Implemented 4 ColliderData algorithms
* Currently have a problem with IsPointInCircle I believe

karnkaul commented on Source/Game/Engine/Physics/ColliderData.cpp:
```diff
@@ -9,25 +9,38 @@ AABBData::AABBData(Vector2 lowerBound, const Vector2 upperBound) : lowerBound(st
+	auto x = point.x - this->centre.x;
+	auto y = point.y - this->centre.y;
+
+	return (this->radius * this->radius) > (x * x) + (y * y);
```
Unfortunately `Fixed` will overflow here, being squared and all; so you'll have to use `ToF64()` / `ToF32()` before squaring each quantity, and compare native floats/doubles that way.

Try this:
```
// IsIntersecting
f64 centreDist = (other.centre - centre).SqrMagnitude();
f64 radiiDist = (other.radius + radius).ToF64();

// IsPointInCircle
f64 distance = (point - centre).SqrMagnitude();
f64 radiusDist = radius.ToF64() * radius.ToF64();
```

cweiss95 replied:
And I can just subtract vectors like that? I saw Vector2 had some operator definitions I think, but I wasn't sure how they exactly worked.

karnkaul replied:
Yeah, add, subtract, Dot, are all supported, and I'd recommend not splitting up a Vector2 unless actually necessary - it's a lot easier to think in terms of them too, esp with orientations!

karnkaul commented:
Quick tip: check those boxes after creating the pull request instead of when writing the description - it's just a mouse click then!

cweiss95 committed:
-Have changed Fixed datatypes to f64 datatypes as when we check
intersections for the circles we square the radius/distance between
objects and this causes Fixed variables to overflow
-Also removed all instances of this
-Added local function InRange() to ColliderData.cpp to make checking
boolean expressions easier
